src/main/java/com/productdock/library/search/book/BookMapper.java

```diff
@@ -1,16 +1,75 @@
 package com.productdock.library.search.book;
 
 import com.productdock.library.search.elastic.document.BookDocument;
+import com.productdock.library.search.elastic.document.Record;
+import com.productdock.library.search.elastic.document.Topic;
+import com.productdock.library.search.kafka.cosumer.messages.BookTopic;
 import com.productdock.library.search.kafka.cosumer.messages.InsertBookMessage;
-import org.mapstruct.Mapper;
-import org.mapstruct.ReportingPolicy;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
-@Mapper(unmappedTargetPolicy = ReportingPolicy.IGNORE, componentModel = "spring")
-public interface BookMapper {
+@Component
+public class BookMapper {
 
-    BookDto toBookDto(BookDocument bookDocument);
+    @Autowired
+    private RecordDtoMapper recordDtoMapper;
 
-    BookDocument toBookDocument(InsertBookMessage insertBookMessage);
+    public BookDto toBookDto(BookDocument bookDocument) {
```

Comment:
Extract till you drop:

In the end, `toBookDto()` should be simple as:
```
        var bookDto = new BookDto();
        mapSimpleProperties(bookDto, bookDocument);
        mapRecords(bookDto, bookDocument.getRecords());
        return bookDto;
```

Try to find other opportunities for extraction.